### DISCUSSION TITLE
alpha clipping + fog

### DISCUSSION THREAD

**Discussion by luisparravicini** [SolarLune/tetra3d#16] [Q&A] [answered]
I've tried to use some textures with alpha clipping and also enable fog and the transparent parts of the textures have the fog applied as they were far away and are obscuring some objects that are behind and near those textures.

I've found that the clip alpha shader (in `camera.go`) had a TODO about discarding the value in certain cases, and modified that to return a color with alpha = 0 when alpha is zero.
I don't know whether that workaround can be considered a generic solution, but for my case it's now working as it should.

The modification is in this commit: https://github.com/luisparravicini/Tetra3d/commit/d43b9df3a7c3be5db9522f08f1c47f6d5fe7de67

**Comment by SolarLune** [comment, marked as answer]
I'm late to reply, but thank you very much for letting me know! I feel like this worked fine without the workaround / addition, but maybe I'm misremembering and just never had it right in the first place, haha. 

Thanks, I've fixed the issue in master / the latest release, and credited you. Regards!

**Reply by luisparravicini** [reply]
Thanks for adding my suggestions.
